### ISSUE TITLE
Update kube-metrics-adapter E2E tests to use autoscaling/v2 API

### ISSUE THREAD

**Issue by lucastt** [zalando-incubator/kubernetes-on-aws#6052]
## What needs to be done and why is it needed?

Change [kube-metrics-adapter e2e test cases](https://github.com/zalando-incubator/kubernetes-on-aws/blob/dev/test/e2e/kube_metrics_adapter_test.go) in kubernetes-on-aws to use `autoscaling/v2` instead of `autoscaling/v2beta1`. All [kube-metrics-adapter metrics already support `autoscaling/v2`](https://github.com/zalando-incubator/kube-metrics-adapter/commit/c4a1e08fdfbbbd94b39c92c0ecd628959a0d8c4b) and this change will be required from Kubernetes version 1.26 onwards.

## What needs to be changed?

Basically change [this import](https://github.com/zalando-incubator/kubernetes-on-aws/blob/956a4196236ccf0700ca2f3b2fb925eac6a9f023/test/e2e/kube_metrics_adapter_test.go#L15) to autoscaling "k8s.io/api/autoscaling/v2" and fix all errors that might arise from this change.

## Acceptance criteria

All E2E tests keep working as usual, but using autoscaling/v2 instead of autoscaling/v2beta2

## Impact

For the E2E tests to run properly Kubernetes version 1.23+ will be needed.
